doc/guides/modules.rst

```diff
@@ -487,6 +505,55 @@ variables. For example:
 
   include(${ZEPHYR_CURRENT_MODULE_DIR}/cmake/code.cmake)
 
+.. _modules_module_root:
+
+Module root
+===========
+
+Zephyr repository contains build files certain known Zephyr modules.
+
+You can add additional paths with build files for Zephyr module by using the
+``-DMODULE_ROOT`` parameter to the CMake build system. This will tell the build
+system to also search this particular path for Zephyr modules using
+``cmake-extern: True`` or ``kconfig-extern: True`` settings as described in
+:ref:`modules_build_files_extern`.
+
+Each additional ``MODULE_ROOT`` must contain a :file:`modules.cmake` file which
+must contain the logic that specifies the build files for Zephyr modules.
+
+The file must be located as:
+
+.. code-block:: none
+
+   <MODULE_ROOT>
+   └── modules
+       └── modules.cmake
+
+
+In the Zephyr repository the build files for Zephyr modules are given names
+according to the following pattern:
+
+.. code-block:: none
+
+   <ZEPHYR_BASE>
+   └── modules
+       └── <module_name>
+           ├── CMakeLists.txt
+           └── Kconfig.<module_name>
+
+For user defined ``MODULE_ROOT`` it is advised to follow a similar structure,
+but you may decide to use a custom pattern.
+A custom pattern can be implemented in the
+:file:`<MODULE_ROOT>/modules/modules.cmake` file.
```

Comment:
>Each additional ``MODULE_ROOT`` must contain a :file:`modules.cmake`

IMO implies

```suggestion
:file:`<MODULE_ROOT>/modules.cmake` file.
```

Reply:
Section description updated to make this clearer.